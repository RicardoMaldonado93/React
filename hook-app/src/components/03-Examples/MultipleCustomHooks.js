import React, {useState} from 'react'
import { useCounter } from '../hooks/useCounter';
import { useFetch } from '../hooks/useFetch'
    
    export const MultipleCustomHooks = () => {

        const { increment, counter } = useCounter(1);
        const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
        const { author, quote } = !!data && data[0];
        
        return (
            <>
                <h1>BreakingBad Quotes</h1>   
                <hr />
                    {
                        loading 
                        ? 
                            ( 
                                <div className="alert alert-info text-center">
                                    loading...
                                </div> 
                            ) 
                        :
                            (
                                <blockquote className="blockquote text-right">
                                    <p className="mb-0">{quote}</p>
                            <footer className="blockquote-footer">{author}</footer>
                                </blockquote>
                            )

                    }

                <button 
                    className="btn btn-primary"
                    onClick={ ()=> increment() }
                >
                    Siguiente
                </button>
            </>
        )
    }
    