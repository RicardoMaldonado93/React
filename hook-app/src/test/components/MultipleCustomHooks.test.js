import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../components/03-Examples/MultipleCustomHooks';
import { useFetch } from '../../components/hooks/useFetch';
import { useCounter } from '../../components/hooks/useCounter';

jest.mock('../../components/hooks/useFetch.js');
jest.mock('../../components/hooks/useCounter.js');

describe('Unitary test with <MultipleCustomHooks />', () => {
    
    test('should be correctly show ', () => {

        useCounter.mockReturnValue({
            counter: 1,
            imcrement: ()=>{}
        })

        useFetch.mockReturnValue({
            data:null,
            loading: true,
            error:null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper ).toMatchSnapshot();
    });
    
    
    test('should be shown information', () => {

        useFetch.mockReturnValue({
            data:[{
                author: "Ricardo",
                quote: "Hola mundo"
            }],
            loading: false,
            error: null
        });


        const wrapper = shallow( <MultipleCustomHooks /> );


        expect( wrapper.find('.alert').exists()).toBeFalsy();
        expect( wrapper.find('.mb-0').text().trim() ).toBe("Hola mundo");
        expect( wrapper.find('footer').text().trim()).toBe("Ricardo");

    });


    
})
