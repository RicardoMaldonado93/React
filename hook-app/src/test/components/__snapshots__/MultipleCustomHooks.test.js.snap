// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary test with <MultipleCustomHooks /> should be correctly show  1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</Fragment>
`;
